packages/fe-fpm-writer/test/integration/index.test.ts

```diff
@@ -147,6 +147,89 @@ describe('use FPM with existing apps', () => {
+                                                    annotationPath: 'com.sap.vocabularies.UI.v1.LineItem'
+                                                },
+                                                {
+                                                    key: 'FirstCustomView',
+                                                    template: 'any.NewCustomView'
+                                                }
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+            generateCustomView(
+                config.path,
+                {
+                    target: 'TravelListReport',
+                    key: 'SecondCustomView',
+                    label: 'Custom View',
+                    name: 'NewCustomView',
+                    eventHandler: true,
+                    ...config.settings
+                },
+                fs
+            );
+        });
+        test.each(configs)('generateCustomView in ListReport, custom view to be overwritten', (config) => {
+            //pre-requisite is at least one view based on annotations
+            fs.extendJSON(join(config.path, 'webapp/manifest.json'), {
+                'sap.ui5': {
+                    routing: {
+                        targets: {
+                            TravelListReport: {
+                                options: {
+                                    settings: {
+                                        views: {
+                                            paths: [
+                                                {
+                                                    key: 'LineItemView',
+                                                    annotationPath: 'com.sap.vocabularies.UI.v1.LineItem'
+                                                },
+                                                {
+                                                    key: 'CustomViewKey',
+                                                    template: 'any.NewCustomView'
+                                                }
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+            generateCustomView(
+                config.path,
+                {
+                    target: 'TravelListReport',
+                    key: 'CustomViewKey',
+                    label: 'Custom View',
+                    name: 'NewCustomView',
+                    eventHandler: true,
+                    ...config.settings
+                },
+                fs,
+                false
+            );
+        });
```

Comment:
In case of integration tests - I think by adding `fs.extendJSON(join(config.path, 'webapp/manifest.json'), {` on each test we are losing progress from previous.

For example if I remove test `generateCustomView in ListReport, second custom view`, then integration test will not fail.

Can we use `generateCustomView in ListReport, custom view to be overwritten` right after `generateCustomView in ListReport` - so that we do not need use `fs.extendJSON(join(config.path, 'webapp/manifest.json'` and then trigger 'generateCustomView in ListReport, second custom view' - so that 3rd view will be added to snapshot?

Reply:
@815are Of course I can change the order of tests - this slightly adapts the snapshot, as you indicated. See following commit.
Indeed, the really important tests come with  packages/fe-fpm-writer/test/unit/view.test.ts

Reply:
@stefanschreck 

I applied additional correction to integration tests - d94e9e306443864516f9882c82345c6cabf44623
my main point was about `fs.extendJSON(join(config.path, 'webapp/manifest.json'), {`.
I mean we have 3 tests regarding views update and in each we overwrite result of previous by `fs.extendJSON(join(config.path, 'webapp/manifest.json'), {`.

For example:
1. Test `generateCustomView in ListReport` - it created 2 views `LineItemView` and `CustomViewKey`
2. Test 'generateCustomView in ListReport, custom view to be overwritten' - overwrites manifest with views  `LineItemView` and `CustomViewKey`. In case if 1st test failed, then end expectation will not fail, because overwrite happened. At then end of test we have `LineItemView` and `CustomViewKey`
3. Test `generateCustomView in ListReport, second custom view`. Old result is overwritten by views `LineItemView`, `FirstCustomView` and then test runs by adding additional view.

In result at then end snapshot 1st and 2nd test will not be checked, because of overwrites in 2nd and 3rd.